py3status/py3.py

```diff
@@ -1008,3 +1008,73 @@ def request(self, url, params=None, data=None, headers=None,
                             headers=headers,
                             timeout=timeout,
                             auth=auth)
+
+    def _val_to_bar(self, value, min_val=0, max_val=100):
+        """
+        Create a bar of the appropriate height for each value and adjusts the
+        colors depending on the thresholds.
+        There are 8 possible levels.
+        By default, this assumes values are percentages, but the min/max can be
+        adjusted.
+
+        :param value: value to transform to a bar
+        :param min_val: minimum value possible
+        :param max_val: maximum value possible
+
+        :returns: Composite
+        """
+
+        bars = ['▁', '▂', '▃', '▄', '▅', '▆', '▇', '█']
+
+        idx = int((value - min_val) / (max_val - min_val) * len(bars))
+        if idx == len(bars):
+            idx -= 1
+        full_text = bars[idx]
+        color = self.threshold_get_color(value)
+        return {'full_text': full_text, 'color': color}
+
+    def concurrent_bar_graphs(self, values, min_val=0, max_val=100):
+        """
+        Transforms a list of value into a list of bar Composites.
+
+        :param values: list of values to transform into a bar graph
+        :param min_val: minimum value possible
+        :param max_val: maximum value possible
+
+        :returns: Composite list
+        """
+
+        val_bars = []
+        for val in values:
+            val_bars.append(self._val_to_bar(val,
+                                             min_val=min_val,
+                                             max_val=max_val))
+
+        return self.composite_create(val_bars)
+
+    def history_bar_graph(self, history, value, length=5, default_value=0, min_val=0, max_val=100):
+        """
+        Transform a value into a bar graph with history.
+
+        :param history: history of the variable
+        :param value: value to add to the history
+        :param length: length of the history
+        :param default_value: default value to pad with
+        :param min_val: minimum value possible
+        :param max_val: maximum value possible
+        """
+
+        if default_value is not None and length is not None:
```

Comment:
Can you explain this part? Could `default` and `length` be `None`? The only way those get `None` is if we give them one... What will happen if we give them `None`? Same thing for `if length is not None:` below.

I think it makes sense if we defaults them `0` and `1`, then we don't have to worry about parameters on other modules if we add them... eg single value = single bar. Maybe that's the plan already. Idk.

Reply:
This is just in case something goes wrong (i.e. someone sets them to `None`). It doesn't make much sense, TBH. I agree with the default to `0` and `1`.